### PR TITLE
feat(j2cl): wire version history overlay

### LINKED ISSUE #1054: F-2: Wire J2CL version-history overlay to WaveletProvider.getHistory + flip restoreEnabled
## Goal

The F-2.S4 (#1048, PR #1052) version-history overlay (`<wavy-version-history>`) ships with a no-op `versionLoader` callback and `restoreEnabled` defaulting to `false`. The overlay chrome (slider + Show changes / Text only / Exit toggles + the inline confirm dialog scaffold) all work; the destructive Restore button is gated until this issue lands.

## Scope

1. Wire the J2CL controller to set the `versionLoader` callback on the mounted `<wavy-version-history>` element after the wave's wavelet is selected. The callback should pull from the existing GWT version-history seam (`WaveletProvider.getHistory`) without modifying that interface.
2. Flip `restoreEnabled = true` on the element once the loader is wired and the wave-snapshot author/version metadata is verified safe to write back.
3. Wire the `wavy-version-changed` event to render a transient snapshot at the chosen version (read-only initially — Restore is the only destructive path).
4. Wire the `wavy-version-restore-confirmed` event to submit a real Restore op via the existing GWT Restore wire (or document why the J2CL path needs its own).

## Out of scope
- Refactoring the `WaveletProvider.getHistory` interface (file a separate issue if the seam needs work).
- Replacing the inline confirm dialog with a custom modal — the native `<dialog>` is the closest GWT-parity choice.

## Verification
- `sbt -batch j2clLitTest j2clSearchTest j2clProductionBuild` exits 0.
- New per-row sibling parity assertion in the J2CL stage-one fixture: with `versionLoader` wired + `restoreEnabled=true`, the Restore button has no `disabled` attribute and the "Preview only — restore not available" hint is gone.
- Manual verification: open a wave with > 5 versions, drag the slider, confirm the displayed snapshot reflects the selected version, click Restore, confirm dialog shows, click Restore in the dialog, confirm the wave is rewritten to the selected version.

## Parent
- F-2.S4 (#1048, PR #1052)
- F-2 umbrella (#1037)
- Tracker (#904)

### PR DESCRIPTION
**Pull request by vega113**
## Summary

Closes #1054. Refs #904.

- Wire the J2CL/Lit version-history overlay to the existing `/history/.../api/*` `VersionHistoryServlet` endpoints.
- Add real history loading, snapshot preview state, visible loading/error/restore status, and restore gating from `/api/info` plus non-empty history.
- Submit confirmed restore via `/api/restore`, then bridge restore success back to `J2clSelectedWaveController.refreshSelectedWave()` through `wavy-selected-wave-refresh-requested` with a current-wave guard.
- Extend parity E2E coverage so the J2CL version-history action must call `/api/info` and `/api/history`, then match the Restore button gate to `canRestore`.

## Verification

- `cd j2cl/lit && npm test -- --files test/wavy-version-history.test.js test/wave-action-bar-controller.test.js` -> pass (`59 passed`).
- `cd j2cl/lit && npm run build` -> pass.
- `cd wave/src/e2e/j2cl-gwt-parity && npm install && npx tsc --noEmit` -> pass.
- `python3 scripts/assemble-changelog.py && python3 scripts/validate-changelog.py --fragments-dir wave/config/changelog.d --changelog wave/config/changelog.json` -> pass (`assembled 308 entries`, validation passed).
- `sbt --batch Test/compile compile j2clSearchTest` -> pass.
- `git diff --check` -> pass.

## Review

- Self-review found and fixed one edge case: Restore is no longer enabled when `canRestore=true` but `/api/history` returns no selectable versions.
- Claude Opus plan and implementation review attempts were both blocked by provider quota: `You've hit your limit · resets May 2 at 9pm (Asia/Jerusalem)`. This is recorded on #1054/#904; no external approval is claimed.

## Summary by CodeRabbit

* **New Features**
  * Version-history overlay: read-only snapshot previews, slider-driven preview, restore action gated by server permission, visible loading/error/restore status, and targeted refresh after successful restore; action-bar now launches and wires the overlay for the selected wave.

* **Documentation**
  * Added implementation plan detailing wiring, UI states, verification commands, and review steps.

* **Tests**
  * Expanded unit, integration, browser, and e2e coverage for loader, preview, restore flows, stale-response handling, and parity.

* **Chores**
  * Added changelog entry.

### CODE REVIEW

File: docs/superpowers/plans/2026-04-30-issue-1054-version-history-wiring.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+**Goal:** Make the J2CL/Lit version-history overlay functional by loading real history data, previewing selected historical snapshots, enabling restore only when the server allows it, and submitting restore through the existing server endpoint.
+
+**Architecture:** Reuse the existing Jakarta `VersionHistoryServlet` instead of adding a new backend path. That servlet already gates access, calls `WaveletProvider.getHistory` for delta metadata, reconstructs snapshots, and submits content-only restore operations. The J2CL/Lit action-bar controller already owns the `wave-nav-version-history-requested` click and opens `<wavy-version-history>`; extend that controller to wire `versionLoader`, `restoreEnabled`, `wavy-version-changed`, and `wavy-version-restore-confirmed` against `/history/{waveDomain}/{waveId}/{waveletDomain}/{waveletId}/api/*`.
+
+---
+
+## Constraints
+
+- Work only in `/Users/vega/devroot/worktrees/issue-1054-version-history-wiring-20260430`.
```
[the remark this answers is not in the document]
Comment (vega113) [reply docs/superpowers/plans/2026-04-30-issue-1054-version-history-wiring.md]: Fixed in d4c8ecc93: made the plan instruction worktree-relative instead of machine-specific.

File: j2cl/lit/src/elements/wavy-version-history.js [hunk 2]
```diff
@@ -431,6 +502,15 @@ export class WavyVersionHistory extends LitElement {
...
         <h2>Version history</h2>
+        ${statusText
+          ? html`<p
+              class="history-status"
+              role=${this.error ? "alert" : "status"}
+            >${statusText}</p>`
+          : null}
+        <p class="current-version-label">
+          ${current ? `Current preview: ${current.label}` : "No versions loaded yet"}
+        </p>
```
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/lit/src/elements/wavy-version-history.js:525]: Fixed in d4c8ecc93: the shared loading status now uses a generic "Loading…" label so snapshot fetches are not described as history loading.

File: j2cl/lit/src/controllers/wave-action-bar-controller.js [hunk 3]
```diff
@@ -264,6 +265,209 @@ function buildFolderUrl(params) {
...
+    if (version == null || !isCurrentWave(host, waveId)) return;
+    overlay.loading = true;
+    overlay.error = "";
+    try {
+      const snapshot = await fetchJson(
+        buildHistoryUrl(apiBase, "/api/snapshot", { version })
+      );
+      if (!isCurrentWave(host, waveId)) return;
+      overlay.snapshot = snapshot;
+      overlay.error = "";
```
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/lit/src/controllers/wave-action-bar-controller.js:420]: Fixed in d4c8ecc93: snapshot preview is cleared before each snapshot fetch, and the controller test now covers the stale-preview case.

File: wave/src/e2e/j2cl-gwt-parity/tests/wave-actions-parity.spec.ts [hunk 4]
```diff
@@ -247,8 +247,27 @@ test.describe("G-PORT-8 top-of-wave action bar parity", () => {
...
+      (response) =>
+        response.url().includes("/history/") &&
+        response.url().includes("/api/info") &&
+        response.status() < 500
+    );
+    const historyResponse = page.waitForResponse(
+      (response) =>
+        response.url().includes("/history/") &&
+        response.url().includes("/api/history") &&
+        response.status() < 500
```
[the remark this answers is not in the document]
Comment (vega113) [reply wave/src/e2e/j2cl-gwt-parity/tests/wave-actions-parity.spec.ts]: Fixed in d4c8ecc93: both parity response predicates now require response.ok() before consuming the API responses.

File: wave/config/changelog.d/2026-04-30-j2cl-version-history-wiring.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
+  "releaseId": "2026-04-30-j2cl-version-history-wiring",
+  "version": "PR #TBD",
```
[the remark this answers is not in the document]
Comment (vega113) [reply wave/config/changelog.d/2026-04-30-j2cl-version-history-wiring.json]: Fixed in d4c8ecc93: changelog fragment now records PR #1143.

File: j2cl/lit/src/controllers/wave-action-bar-controller.js [hunk 6]
```diff
@@ -264,6 +265,209 @@ function buildFolderUrl(params) {
...
+    } catch (err) {
+      if (isCurrentWave(host, waveId)) {
+        overlay.error = messageFromError(err, "Unable to load version snapshot.");
+      }
+    } finally {
+      if (isCurrentWave(host, waveId)) {
+        overlay.loading = false;
+      }
+    }
```
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/lit/src/controllers/wave-action-bar-controller.js:430]: Fixed in 2f8011bfc: snapshot fetches now use a latest-request token, clear the preview before loading, and ignore stale success/error/finally callbacks. Added out-of-order snapshot coverage.

File: j2cl/lit/src/controllers/wave-action-bar-controller.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/lit/src/controllers/wave-action-bar-controller.js]: Fixed in 2f8011bfc: older snapshot responses can no longer overwrite the preview after the user moves to a newer selection.

File: j2cl/lit/src/controllers/wave-action-bar-controller.js [hunk 7]
```diff
@@ -264,6 +265,210 @@ function buildFolderUrl(params) {
...
+          }
+        })
+      );
+    } catch (err) {
+      if (isCurrentWave(host, waveId)) {
+        overlay.restoreStatus = "";
+        overlay.error = messageFromError(err, "Unable to restore version.");
+      }
+    }
+  };
```
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/lit/src/controllers/wave-action-bar-controller.js:482]: Fixed in 4b8158f35: restore confirmations now use an in-flight guard, disable Restore while the POST is pending, and re-enable it after completion for the current wave. Added duplicate-submit coverage.

File: wave/src/e2e/j2cl-gwt-parity/tests/wave-actions-parity.spec.ts [hunk 8]
```diff
@@ -247,8 +247,27 @@ test.describe("G-PORT-8 top-of-wave action bar parity", () => {
...
+    );
     await clickActionJ2cl(page, "version-history");
+    const info = await (await infoResponse).json();
+    await historyResponse;
     await expect(overlay).not.toHaveAttribute("hidden", "");
+    await expect(overlay.locator(".current-version-label")).toBeVisible();
+    await expect(overlay.locator("button.restore")).toHaveAttribute(
+      "aria-disabled",
+      info.canRestore ? "false" : "true"
+    );
```
[the remark this answers is not in the document]
Comment (vega113) [reply wave/src/e2e/j2cl-gwt-parity/tests/wave-actions-parity.spec.ts:270]: Fixed in 4b8158f35: the parity assertion now uses the same gate as the controller, `canRestore && history.length > 0`.
